DESIGN.md

```diff
@@ -0,0 +1,67 @@
+
+
+# Design Principles - Addressed
+
+This is a rough accounting of the design problems addressed so far.
+
+### Axis Mundi
+
+The Postgres database is the center of the world and the application design flows outward.
+
+- We want the database to maintain as much consistency as possible, such that any program interacting with the database (not just the new ODB) will see an accurate view of the world and will only be able to make modifications that are "legal" under some definition of well-formedness.
+- We want the database to be queryable easily, such that a read-replica can be made available to science staff for their own use. To this end some *checked* denormalization (such as program ids being broken into site, year, etc.) is ok.
+- Similarly astronomical units need to be in a science-friendly format, and we may need server-side functions for doing computations. We need to be cautious about pushing too much *business* logic into the database but support for domain-specific data types is probably reasonable.
+- Enumerated types like filters, dispersers, etc., are important for reporting and consistency checking and need to be defined in the database. We use code-generation to read these in at compile time and make them available as proper enumerated types in our Scala code.
+- Events should be logged to the database and echoed to traditional file logs in case database connectivity fails.
```

Comment:
Traditionally logs don't go to the db due to the high cost of writing many entries to the db. How do you see that? I'm thinking we may want to log user interactions, like user X added node Y, which could be a fair amount of traffic